### PR TITLE
catch2: modernize more

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jwillikers — DISMISSED** [review]

File: recipes/catch2/3.x.x/conandata.yml [hunk 1]
```diff
@@ -5,8 +5,3 @@ sources:
   "3.0.1":
     url: "https://github.com/catchorg/Catch2/archive/v3.0.1.tar.gz"
     sha256: "8c4173c68ae7da1b5b505194a0c2d6f1b2aef4ec1e3e7463bde451f26bbaf4e7"
-patches:
-  "3.1.0":
-    - patch_file: "patches/3.0.1-0001-force-not_usbproject-off.patch"
```
Comment (prince-chrismc) [inline recipes/catch2/3.x.x/conandata.yml:10]: 🚀

File: recipes/catch2/3.x.x/conanfile.py [hunk 2]
```diff
@@ -17,111 +18,109 @@ class Catch2Conan(ConanFile):
     url = "https://github.com/conan-io/conan-center-index"
     settings = "os", "arch", "compiler", "build_type"
     options = {
+        "shared": [True, False],
         "fPIC": [True, False],
         "with_prefix": [True, False],
-        "default_reporter": "ANY",
```
Comment (prince-chrismc) [inline recipes/catch2/3.x.x/conanfile.py:22]: This option still exists 😕 https://github.com/catchorg/Catch2/blob/c0e582e65955f84312548bcee36b962a013f262f/CMake/CatchConfigOptions.cmake#L72

https://github.com/catchorg/Catch2/blob/c0e582e65955f84312548bcee36b962a013f262f/docs/configuration.md#default-reporter

I think this should be kept

Comment (SpaceIm) [reply recipes/catch2/3.x.x/conanfile.py:22]: indeed

**Comment by SpaceIm** [comment]
> Please consider the following example taken from Catch2 documentation: https://github.com/catchorg/Catch2/blob/devel/docs/cmake-integration.md#usage
> 
> It should work using the Conan Catch2 recipe:
> 
> ```
> cmake_minimum_required(VERSION 3.5)
> 
> project(baz LANGUAGES CXX VERSION 0.0.1)
> 
> find_package(Catch2 REQUIRED)
> add_executable(foo test.cpp)
> target_link_libraries(foo PRIVATE Catch2::Catch2)
> 
> include(CTest)
> include(Catch)
> catch_discover_tests(foo)
> ```
> 
> With the current recipe I have issues with `include(Catch)`

Ok, let's test this in test_package, and fix builddirs.

**Comment by prince-chrismc** [comment]
To fix the build you need to more the "delete option" to `configure` but you can check the issue referenced from the main repo https://github.com/conan-io/conan/issues/12211

❤️ @jgsogo who helped me understand

**Comment by SpaceIm** [comment]
Yes. The issue is that for several conan commands, an option deleted in config_options is not exactly the same than a non-existent option.

**Review by jwillikers — APPROVED** [review]

**Comment by SpaceIm** [comment]
and if ["ANY"] is replace by [None, "ANY"]?

**Comment by uilianries** [comment]
> and if ["ANY"] is replaced by [None, "ANY"]?

You still can use `None` when option is `["ANY"]`

**Comment by uilianries** [comment]
So it's a Conan v2 bug, or not documented. Even the documentation is very clear about ANY option: https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#any-special-value

**Comment by prince-chrismc** [comment]
Open an issue to the main repo :)

**Comment by SpaceIm** [comment]
https://github.com/conan-io/conan-center-index/pull/13240

File: recipes/catch2/3.x.x/conanfile.py [hunk 3]
```diff
@@ -17,60 +18,71 @@ class Catch2Conan(ConanFile):
...
         get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
 
     def generate(self):
         tc = CMakeToolchain(self)
         tc.variables["BUILD_TESTING"] = False
-        tc.variables["CATCH_INSTALL_DOCS"] = False
-        tc.variables["CATCH_INSTALL_HELPERS"] = True
+        tc.cache_variables["CATCH_INSTALL_DOCS"] = False
+        tc.cache_variables["CATCH_INSTALL_EXTRAS"] = True
+        tc.cache_variables["CATCH_DEVELOPMENT_BUILD"] = False
```
Comment (paulharris) [inline recipes/catch2/3.x.x/conanfile.py:85]: @SpaceIm can you please educate me why you chose to use cache_variables and not variables?

**Pull request merged**